src/main/java/org/wildfly/security/WildFlyElytronProvider.java

```diff
@@ -223,4 +229,56 @@ private void putCredentialStoreProviderImplementations() {
         putService(new Service(this, CredentialStore.CREDENTIAL_STORE_TYPE, MaskedPasswordStore.MASKED_PASSWORD_STORE, MaskedPasswordStore.class.getName(), emptyList, emptyMap));
     }
 
+    static class ProviderService extends Service {
+
+        private volatile Reference<Class<?>> classReference;
```

Comment:
There's no reason to make this be a special Reference because AFAICT all the classes loaded by this provider will also be defined by the same class loader as this class.  Rather than caching one of these, it might be better to pass in a Supplier or ExceptionSupplier which can instantiate the thingy directly, using a constructor method reference for the value.  But at least we should not have this.